include/boost/compute/container/valarray.hpp

```diff
@@ -90,7 +90,30 @@ class valarray
     ~valarray()
     {
     }
-
+ 
+    valarray<T> operator+(const valarray<T> &rhs) const
++    {
```

Comment:
Extra '+' symbol seems to be the problem... Same thing in a few more lines that follow...